src/pages/viewCart/Cart.tsx

```diff
@@ -209,9 +214,33 @@ function CartTable() {
                 </Link>
               </div>
             </div>
+            {deleteWarning && (
+              <PopUpMaker
+                Component={WarningDelete}
+                setIsPopupOpen={setDeleteWarning}
+                handleGetCart={handleGetCart}
```

Comment:
This card is not needed here, if it is required there in PopUp go make it optional there in popUp, and then make render the card items on condition (if the card exists then render it) by this method, you are saving almost all those 30 lines you are repeating again